standard_rules/unusual_login.py

```diff
@@ -96,18 +101,52 @@ def rule(event):
     # Mocked during unit testing
     put_string_set(event_key, [json.dumps(updated_geo_logins)])
 
+    global GEO_HISTORY  # pylint: disable=global-statement
+    GEO_HISTORY = updated_geo_logins
+    logging.debug("GEO_HISTORY in main rule:\n%s", json.dumps(GEO_HISTORY))
     return True
 
 
-def get_key(event):
+def get_key(event) -> str:
     # Use the name to deconflict with other rules that may also use actor_user
     return __name__ + ":" + str(event.udm("actor_user"))
 
 
 def title(event):
+    event_time_truncated = nano_to_micro(event.get("p_event_time"))
+    parse_time_truncated = nano_to_micro(event.get("p_parse_time"))
+    time_delta = resolve_timestamp_string(parse_time_truncated) - resolve_timestamp_string(
+        event_time_truncated
+    )
+    days = time_delta.days
+    hours, remainder = divmod(time_delta.seconds, 3600)
+    minutes, seconds = divmod(remainder, 60)
+    parsing_delay = ""
+    if days > 0:
+        parsing_delay = f"{days} day(s) "
+    if hours > 0:
+        parsing_delay = "".join([parsing_delay, f"{hours} hour(s) "])
+    if minutes > 0:
+        parsing_delay = "".join([parsing_delay, f"{minutes} minute(s) "])
+    if seconds > 0:
+        parsing_delay = "".join([parsing_delay, f"{seconds} second(s)"])
+
     return (
         f"{event.get('p_log_type')}: New access location for user"
         f" [{event.udm('actor_user')}]"
         f" from {GEO_INFO.get('city')}, {GEO_INFO.get('region')} in {GEO_INFO.get('country')}"
-        f" (not in last [{GEO_HISTORY_LENGTH}] login locations)"
+        f" (not in last [{GEO_HISTORY_LENGTH}] login locations)\n"
+        f"Parsing delay: {parsing_delay}"
```

Comment:
I think this would be better in the `alert_context` function, it looks a bit verbose for the `title` which needs to generically  fit in e.g. a slack message, an email subject line, an asana/jira ticket, etc.

Reply:
I think it's important to have the info immediately at hand in the original alert (also one less thing to have to learn for new people, or people who don't see this rule trigger very often). I have my screen set on a super low font resolution, and adding that in at the end (and on a new line for destinations that support it) wouldn't add that much, even in Asana. Slack shows the entire title regardless of how long it is.

If the destination functionality could put `alert_context` info into the alert body, I'd agree with your suggestion.

Reply:
I also like the idea of including this in the alert context rather than the title.

But also... if this is useful information for this rule, isn't it useful for most (if not all) detections (particularly any relating to gsuite)? If that is the case, another option could be to add something like `add_parse_delay(event, context)` method that takes in an alert context dictionary and the raw event - returns the alert context with parse delay added to it?

Reply:
Another way of thinking about this (which I just thought of): I think `alert_context` is better situated to displaying machine-readable-ish data, like an actual `timedelta()` object string, vs the human-readable processed form I have now. If anyone hypothetically wanted to, say, compute the mean/std deviation of parsing delays for logs, I have a vague feeling it would be easier to do if we had `alert_context` = `{"parsingDelay": "datetime.timedelta(days=1, seconds=15, microseconds=930269)"}` or `{"parsingDelay": "1 day, 1:00:15.930269"}` vs the days/hours/minutes/seconds human readable (and variable) text string.

Reply:
> If that is the case, another option could be to add something like `add_parse_delay(event, context)` method that takes in an alert context dictionary and the raw event - returns the alert context with parse delay added to it?

Agreed. I'll add a helper.